server/client_rest.go

```diff
@@ -70,6 +71,27 @@ func (client *clientRest) NodeSendStats(nodeKey string, sessionList []dto.Sessio
 	return nil
 }
 
+func (client *clientRest) Proposals(nodeKey string) (proposals []dto_discovery.ServiceProposal, err error) {
+	response, err := client.doRequest("GET", "proposals", dto.ProposalsRequest{
+		NodeKey: nodeKey,
+	})
+
+	if err == nil {
```

Comment:
Let's do the following way:
```if err != nil {
    return
}
noErrorLogic()```

Reply:
This looked strange for me as well, but there are a lot such cases in this file!
This fix this instance, but other instances should be refactored in a separate PR not to explode this one.

Reply:
Done.